drizzlepac/haputils/cell_utils.py

```diff
@@ -540,70 +539,53 @@ def find_corners(self, member='total'):
                 label_shape = (mask_slice[0].stop - mask_slice[0].start + (label_border * 2),
                                mask_slice[1].stop - mask_slice[1].start + (label_border * 2))
-                label_mask = np.zeros(label_shape, sclabels.dtype)
+                # create a boolean version of mask for this label
+                label_mask = np.zeros(label_shape, dtype=bool)
                 # get slice with just the region/label of interest
-                label_mask[label_border:-1*label_border, label_border:-1*label_border] = sclabels[mask_slice].copy()
-                # make sure no pixels from other regions are present in this mask
-                label_mask[label_mask != label] = 0
-                # reset label to be a binary mask only
-                label_mask[label_mask == label] = 1000
-                print('extracting corners for region {} in slice {}'.format(label, mask_slice))
-                # Perform corner detection on each region/chip separately.
-                mask_corners = corner_peaks(corner_harris(label_mask),
-                                       min_distance=3,
-                                       threshold_rel=0.2)
-                xy_corners = mask_corners * 0.
-                xy_corners[:, 0] = mask_corners[:, 1]
-                xy_corners[:, 1] = mask_corners[:, 0]
-                # shift corner positions to full array positions
-                xy_corners += (mask_slice[1].start-label_border, mask_slice[0].start-label_border)
+                label_mask[label_border:-1*label_border, label_border:-1*label_border] = (sclabels[mask_slice] == label)
+
+                # Perform edge tracing on each region/chip separately.
 
                 # Create a mask from the total footprint consisting solely of the
                 # pixels at the outer edge, ordered in clockwise fashion.
-                #
-                # get list of (X,Y) coordinates of all edges from each separate 'region' or chip
-                edge_pixels = trace_polygon(label_mask > 0, mask_slice)
-
-                # use the ordering of the traced edge pixels to order the corners in the same way
-                cordist = distance.cdist(xy_corners, edge_pixels)  # returns distances for each corner position
-                ordered_indices = []
-                for distarr, minval in zip(cordist, np.min(cordist, axis=1)):
-                    ordered_indices.append(np.where(distarr == minval)[0][0])
-                radial_order = np.argsort(ordered_indices)
-                ordered_xyc = xy_corners[radial_order].tolist()
-                ordered_xyc.append(ordered_xyc[0])  # close polygon
+                # Correct for the label_border padding
+                edge_pixels = trace_polygon(label_mask, mask_slice) - label_border
+
+                # simplify the polygon
+                # note edge_pixels and xy_corners polygons are already closed
+                xy_corners = simplify(edge_pixels, min_dist=3.0)
```

Comment:
```suggestion
                xy_corners = simplify(edge_pixels, min_dist=10.0)
```
Later testing shows that `min_dist=10.0` is a better simplification threshold.  It reduces the complexity of the footprints a bit.